src/lib/mina_base/transaction.ml

```diff
@@ -78,8 +78,8 @@ let forget : Valid.t -> t = fun x -> (x :> t)
 let fee_excess : t -> Fee_excess.t Or_error.t = function
   | Command (Signed_command t) ->
       Ok (Signed_command.fee_excess t)
-  | Command (Parties _ps) ->
-      failwith "TODO"
+  | Command (Parties ps) ->
+      Ok (Parties.fee_excess ps)
```

Comment:
Turn this back into a `failwith`: it's not valid to call this function on a parties transaction.

It would be better to remove this function altogether and return the other necessary data (fee excess, etc.) from `Ledger.apply_transaction`, so that the accurate state is available to `Staged_ledger`.